apps/gallery/js/gallery.js

```diff
@@ -493,8 +501,13 @@ function initThumbnails() {
     PerformanceTestingHelper.dispatch('media-enumerated');
 
     // Now that we've enumerated all the photos and videos we already know
-    // about go start looking for new photos and videos.
-    photodb.scan();
+    // about it is time to go and scan the filesystem for new ones. If the
+    // MediaDB is fully ready, we can scan now.  Either way, we always want
+    // to scan every time we get a new 'ready' event.
+    photodb.addEventListener('ready', function() { photodb.scan(); });
```

Comment:
How does this new event listener change the behavior of the gallery app? I'm guessing it fires whenever you unplug USB. Did Gallery just not handle that case before?